plugins/modules/packaging/language/maven_artifact.py

```diff
@@ -509,7 +520,21 @@ def _request(self, url, failmsg, force=True):
         self.module.params['url_password'] = self.module.params.get('password', '')
         self.module.params['http_agent'] = self.user_agent
 
-        response, info = fetch_url(self.module, url_to_use, timeout=req_timeout, headers=self.headers)
+        if parse_version(ansible_version) <= parse_version("2.11.*"):
```

Comment:
Here we check whether we can use `unredirected_headers` or not based on the version.
Should I also add something more for `2.11` users? 

Something like an Exception, if they are setting the headers but cannot used them because of their version of ansible, or something like that.

Not sure if the behaviour of ignoring that field for 2.11 is desired or not.

Reply:
Use `from ansible_collections.community.general.plugins.module_utils.version import LooseVersion` and then
```suggestion
        if LooseVersion(ansible_version) < LooseVersion("2.12"):
```

Reply:
> Should I also add something more for `2.11` users?
> 
> Something like an Exception, if they are setting the headers but cannot used them because of their version of ansible, or something like that.

You should handle that directly after `AnsibleModule` has been created. Use `module.fail_json(msg=...)` to fail the module if `unredirected_headers` has been passed and `ansible_version` is before 2.12.